app/packages/core/src/components/Filters/use-query-performance-icon.tsx

```diff
@@ -0,0 +1,45 @@
+import * as fos from "@fiftyone/state";
+import React from "react";
+import { useRecoilValue } from "recoil";
+import { LightningBolt } from "../Sidebar/Entries/FilterablePathEntry/Icon";
+
+const Icon = ({ path }: { path: string }) => {
+  const hasFilters = useRecoilValue(fos.hasFilters(false));
+  const filteredIndex = useRecoilValue(
+    fos.pathHasIndexes({ path, withFilters: true })
+  );
+  const color = useRecoilValue(fos.pathColor(path));
+
+  return (
+    <LightningBolt
+      color={filteredIndex ? color : undefined}
+      tooltip={filteredIndex && hasFilters ? "Compound index" : "Indexed"}
```

Comment:
@benjaminpkane -- Isn't this the tooltip where we wanted to be more descriptive?